docassemble/ALWeaver/data/questions/template_validation.yml

```diff
@@ -562,6 +565,30 @@ buttons:
       code: |
         warn_pdf_variable_names_in_docx = True
   - Restart: restart
+---    
+id: did you intend to use reserved variable names
+question: |
+  Did you intend to use variables that have a reserved meaning in Docassemble interviews?
+subquestion: |
+  Your DOCX template looks like it is using a variable name that has a
+  reserved meaning. You won't be asked a question to define this variable in your interview.
+  It will get the value or use the meaning that it normally has in Docassemble.
```

Comment:
This is a good explanation, but it's not complete IMO. Specifically, looking over https://suffolklitlab.org/docassemble-AssemblyLine-documentation/docs/framework/reserved_keywords/, there are a handful of keywords that are:

* python keywords, so will likely error out the jinja template if they haven't already
* Classes or functions, that are 100% not intended to be placed directly by value in a document

So, "the meaning that it normally has in Docassemble" of those values are certainly not the meanings that most users of the Weaver want to use.

Not sure how the fields used are gotten from DOCX files. If `ordinal` as a raw function is in a jinja template, that's probably not intended, but if something like `ordinal(idx)` is listed as `ordinal` in the fields, I could see how the current screen is fine.

Reply:
I just don't think we can correctly decide when people are misusing a reserved keyword in a DOCX template. The warning is about as strong as we can get to my mind.

We could try making two lists, maybe, with the ones that would error on their own list, but the recent method of just blocking everything is more incorrect than allowing with a warning.

Reply:
Yeah, two lists would be good. Doesn't have to be this PR, but IMO people will skip warnings, especially ones that are kinda vague